### PR TITLE
add security.01

### PR DESCRIPTION
**Pull request by YuraNR**

### CODE REVIEW

**Review by sonyasapsan — APPROVED** [review]

File: src/main/java/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -33,14 +34,15 @@ public class BookController {
     @GetMapping
     @Operation(summary = "Get all books",
             description = "Get a list of all books")
-    public List<BookDto> getAll(@PageableDefault(page = 0, size = 2) Pageable pageable) {
+    public List<BookDto> getAll(@PageableDefault(page = 0, size = 2, direction = Sort.Direction.ASC)
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/controller/BookController.java:37]: You ghouls add @PreAuthorize for endpoints

File: src/main/java/bookstore/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/security/CustomUserDetailsService.java:17]:
```suggestion
    public UserDetails loadUserByUsername(String username) {
```

File: src/main/java/bookstore/security/CustomUserDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/security/CustomUserDetailsService.java:13]: remove empty line

File: src/main/java/bookstore/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.email()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/service/impl/UserServiceImpl.java:24]: It's better to create boolean existsByEmail method

File: src/main/java/bookstore/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            throw new RegistrationException("User with such email already exists");
+        }
+        User user = new User();
+        user.setPassword(passwordEncoder.encode(requestDto.password()));
+        user.setEmail(requestDto.email());
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/service/impl/UserServiceImpl.java:29]:
```suggestion
        user.setRole(Set.of(....
```

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: email
+                  type: varchar(255)
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml:18]: Check all sizes. They should be the same as in requests

File: src/main/java/bookstore/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import bookstore.exception.RegistrationException;
+import bookstore.mapper.UserMapper;
+import bookstore.model.User;
+import bookstore.repository.UserRepository;
+import bookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/service/impl/UserServiceImpl.java:14]:
```suggestion
@Service
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Pull request merged**